File: android-native/app/src/main/java/com/example/kpkn/screens/workout/components/VolumeAdvanceModal.kt
package com.example.kpkn.screens.workout.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.example.kpkn.data.models.MuscleAdvance
import kotlin.math.roundToInt

@Composable
fun VolumeAdvanceModal(
    advances: List<MuscleAdvance>,
    onAccept: () -> Unit,
    onDismiss: () -> Unit,
) {
    ProtectedWorkoutBottomSheet(
        title = "Adelanto de volumen",
        onDismiss = onDismiss,
        showCloseButton = false,
    ) {
        Text(
            text = "Realizaste más series de las planificadas.",
            style = MaterialTheme.typography.bodyMedium,
            color = Color.White.copy(alpha = 0.78f),
        )

        advances.forEach { advance ->
            val surplusSets = advance.deficitSets.roundToInt().coerceAtLeast(1)
            val muscle = advance.muscleName.ifBlank { advance.muscleId }
            Surface(
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(16.dp),
                color = Color(0xFF2A2A2A),
            ) {
                Column(
                    modifier = Modifier.padding(14.dp),
                    verticalArrangement = Arrangement.spacedBy(8.dp),
                ) {
                    Text(
                        text = muscle,
                        style = MaterialTheme.typography.titleSmall,
                        fontWeight = FontWeight.Black,
                        color = Color.White,
                    )
                    Text(
                        text = "Si sumamos estas a las programadas para la siguiente sesión que involucra $muscle, superas el volumen. ¿Compensas restando de esa sesión las $surplusSets series extra?",
                        style = MaterialTheme.typography.bodySmall,
                        color = Color.White.copy(alpha = 0.82f),
                    )
                    Text(
                        text = "Próxima sesión: ${advance.targetSessionName}",
                        style = MaterialTheme.typography.bodySmall,
                        color = Color.White.copy(alpha = 0.65f),
                    )
                    advance.discountProposals.forEach { proposal ->
                        val discount = proposal.discountSets.roundToInt().coerceAtLeast(1)
                        Text(
                            text = "• ${proposal.exerciseName}: −$discount ${if (discount == 1) "serie" else "series"}",
                            style = MaterialTheme.typography.bodySmall,
                            color = Color.White.copy(alpha = 0.85f),
                        )
                    }
                }
            }
        }

        Spacer(Modifier.height(8.dp))

        Button(
            onClick = onAccept,
            modifier = Modifier.fillMaxWidth(),
            shape = RoundedCornerShape(14.dp),
            colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.primary),
        ) {
            Text("Aplicar descuento", fontWeight = FontWeight.Black)
        }

        OutlinedButton(
            onClick = onDismiss,
            modifier = Modifier.fillMaxWidth(),
            shape = RoundedCornerShape(14.dp),
        ) {
            Text("Omitir por ahora", color = Color.White.copy(alpha = 0.85f))
        }
    }
}

File: ios-native/KPKNFit/KPKNFit/Presentation/Screens/Workout/Components/VolumeAdvanceModal.swift
import SwiftUI

struct VolumeAdvanceModal: View {
    let advances: [MuscleAdvance]
    let onAccept: () -> Void
    let onDismiss: () -> Void

    var body: some View {
        ProtectedWorkoutBottomSheet(title: "Adelanto de volumen", onDismiss: onDismiss, showCloseButton: false) {
            Text("Completaste más series de las planificadas. Puedes descontar ese volumen en la próxima sesión de la semana que use los mismos músculos.")
                .font(.subheadline)
                .foregroundColor(.white.opacity(0.78))

            ForEach(advances) { advance in
                let surplusSets = max(Int(advance.deficitSets.rounded()), 1)
                VStack(spacing: 8) {
                    VStack(alignment: .leading, spacing: 8) {
                        Text(advance.muscleName.isEmpty ? advance.muscleId : advance.muscleName)
                            .font(.system(size: 14, weight: .black))
                            .foregroundColor(.white)
                        Text("+\(surplusSets) series extra en esta sesión")
                            .font(.system(size: 12))
                            .foregroundColor(Color(red: 0.49, green: 0.83, blue: 0.99))
                            .fontWeight(.bold)
                        Text("Próxima sesión: \(advance.targetSessionName)")
                            .font(.system(size: 12))
                            .foregroundColor(.white.opacity(0.65))
                        ForEach(advance.discountProposals) { proposal in
                            let discount = max(Int(proposal.discountSets.rounded()), 1)
                            Text("• \(proposal.exerciseName): −\(discount) \(discount == 1 ? "serie" : "series")")
                                .font(.system(size: 12))
                                .foregroundColor(.white.opacity(0.85))
                        }
                    }
                    .padding(14)
                    .frame(maxWidth: .infinity, alignment: .leading)
                    .background(Color(red: 0.165, green: 0.165, blue: 0.165))
                    .clipShape(RoundedRectangle(cornerRadius: 16))
                }
            }

            Button(action: onAccept) {
                Text("Aplicar descuento")
                    .fontWeight(.black)
                    .frame(maxWidth: .infinity)
            }
            .buttonStyle(.borderedProminent)
            .tint(.blue)
            .clipShape(RoundedRectangle(cornerRadius: 14))

            Button(action: onDismiss) {
                Text("Omitir por ahora")
                    .foregroundColor(.white.opacity(0.85))
                    .frame(maxWidth: .infinity)
            }
            .buttonStyle(.bordered)
            .tint(.white.opacity(0.3))
            .clipShape(RoundedRectangle(cornerRadius: 14))
        }
    }
}
